libs/common.js

```diff
@@ -0,0 +1,183 @@
+const fs = require('fs');
+const path = require('path');
+
+// minute * seconds * milliseconds
+const MSONEHOUR = 60 * 60 * 1000;
+
+function expect() {
+    for (var i=0, max = arguments.length; i < max; i++) {
+        if (arguments[i] !== null && arguments[i] !== undefined)
+            return arguments[i];
+    }
+
+    return null;
+}
+
+function trim(str) {
+	return str.replace(/^\s+|\s+$/g, '');
+}
+
+function utcnow() {
+    return new Date().toISOString();
+}
+
+//* @now = current date
+//* @days = int, number of days before cookie expires
+function getFutureDays(days) {
+    var now = new Date();
+    now.setTime(now.getTime() + (days * 24 * 60 * 60 * 1000));
+    return now;
+}
+
+//* @now = current date
+//* @ms = int, number of milliseconds before cookie expires
+function getFutureMs(ms) {
+    var now = new Date();
+    now.setMilliseconds(now.getMilliseconds() + ms);
+    return now;
+}
+
+function parseJson(str) {
+    try {
+        return JSON.parse(str);
+    } catch (err) {
+        return false;
+    }
+}
+
+function toJson(obj) {
+    try {
+        return JSON.stringify(obj);
+    } catch (err) {
+        return false;
+    }
+}
+
+
+function filterDir(fpath, extension='.json', recursive=true, encoding='utf8') {
+    var list=[];
+
+    fs.readdirSync(fpath, encoding).forEach(file => {
+        var fstat,
+            fname = path.join(fpath, file);
+
+        fstat = fs.statSync(fname);
+        if (!fstat)
+            return;
+
+        if (fstat.isDirectory()) {
+            list = list.concat(filterDir(fname, extension));
+        } else if (fstat.isFile() && path.extname(file) == extension) {
+            list.push(fname);
+        }
+    });
+
+    return list;
+}
+
+function genRandomToken(length=32) {
+    var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789",
+        token = "";
+
+    for (var i = 0; i < length; i++) {
+        token += possible.charAt(Math.floor(Math.random() * possible.length));
+    }
+
+    return token;
+}
+
+function ASYNC(fn) {
```

Comment:
can we use `util.promisify`, we can just target `node@>=8` ?

Reply:
util.promisify performs the same task, but doesn't do it exactly the same way. There is an expectation that errors return fast instead of reject on a promise.

Reply:
If we decide to just remove the expectation then I'll refactor to use util.promisify instead.